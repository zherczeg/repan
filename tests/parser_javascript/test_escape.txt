# Test escape sequences

/\+\?\*\{\^\$\.\\\(\[/J
    print

/\a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z/J
    print

/\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z/J
    print

/()\1\2\0033/J
    print

/\80\90/J
    print

/\x\x1\x112/J
    print

/\ud7ff\ue000\u{10fFf}\ud800\udc00/16uJ
    ascii, print

/\u\u1\u12\u123\u1234\u12345/16uJ
    ascii, print

/\ud800#/16uJ
    print

/\udfff#/16uJ
    print

/\ud800\udc/16uJ
    ascii, print

# Unicode property support

/\pL\p{L}\P{Cc}\p{SignWriting}/Ju
    print

/\pA/Ju
    print

/\p{L}/J
    print

/\p{Unknown}/Ju
    print

/\p{Chak!ma}/Ju
    print

/\p{Chakma/Ju
    print

/\p{Ahom}\P{Cher}\p{Plrd}\P{Miao}/Ju
    print

/\p{AHom}/Ju
    print

/\p{ Cher}/Ju
    print

/\p{sc=Egyp}\P{Script=Hira}\p{scx=Linear_B}\P{Script_Extensions=Multani}/Ju
    print

/\p{gc=Ps}\P{General_Category=Po}/Ju
    print

/\p{sc=Ps}/Ju
    print

/\p{General_Category=Linear_A}/Ju
    print

/\p{scx}/Ju
    print

/\p{gc= Ps}/Ju
    print

/\P{General_Category=Po }/Ju
    print
