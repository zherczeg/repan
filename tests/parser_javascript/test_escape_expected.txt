# Test escape sequences

/\+\?\*\{\^\$\.\\\(\[/J
    print
    /\+\?\*\{\^\$\.\\\(\[/

/\a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z/J
    print
    /a\bc\de\fghijlm\nopq\r\s\tu\x{b}\wxyz/

/\A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z/J
    print
    /A\BC\DEFGHIJKLMNOPQR\STUV\WXYZ/

/()\1\2\0033/J
    print
    /()\g{1}\x{2}\x{3}3/

/\80\90/J
    print
    /880990/

/\x\x1\x112/J
    print
    /xx1\x{11}2/

/\ud7ff\ue000\u{10fFf}\ud800\udc00/16uJ
    ascii, print
    /\x{d7ff}\x{e000}\x{10fff}\x{10000}/

/\u\u1\u12\u123\u1234\u12345/16uJ
    ascii, print
    /uu1u12u123\x{1234}\x{1234}5/

/\ud800#/16uJ
    print
    Pattern parse failed: Invalid UTF character
        /[ HERE ]>>\ud800#/

/\udfff#/16uJ
    print
    Pattern parse failed: Invalid UTF character
        /[ HERE ]>>\udfff#/

/\ud800\udc/16uJ
    ascii, print
    Pattern parse failed: Invalid UTF character
        /[ HERE ]>>\ud800\udc/

# Unicode property support

/\pL\p{L}\P{Cc}\p{SignWriting}/Ju
    print
    /pL\p{L}\P{Cc}\p{SignWriting}/

/\pA/Ju
    print
    /pA/

/\p{L}/J
    print
    Pattern parse failed: UTF support needs to be enabled for this pattern
        /[ HERE ]>>\p{L}/

/\p{Unknown}/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\p{Unknown}/

/\p{Chak!ma}/Ju
    print
    Pattern parse failed: A valid unicode property required after \p or \P
        /[ HERE ]>>\p{Chak!ma}/

/\p{Chakma/Ju
    print
    Pattern parse failed: A right brace (}) is expected
        /\p{Chakma[ HERE ]>>/

/\p{Ahom}\P{Cher}\p{Plrd}\P{Miao}/Ju
    print
    /\p{Ahom}\P{Cherokee}\p{Miao}\P{Miao}/

/\p{AHom}/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\p{AHom}/

/\p{ Cher}/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\p{ Cher}/

/\p{sc=Egyp}\P{Script=Hira}\p{scx=Linear_B}\P{Script_Extensions=Multani}/Ju
    print
    /\p{Egyptian_Hieroglyphs}\P{Hiragana}\p{Script_Extensions=Linear_B}\P{Script_Extensions=Multani}/

/\p{gc=Ps}\P{General_Category=Po}/Ju
    print
    /\p{Ps}\P{Po}/

/\p{sc=Ps}/Ju
    print
    Pattern parse failed: Unicode script name expected
        /\p{sc=[ HERE ]>>Ps}/

/\p{General_Category=Linear_A}/Ju
    print
    Pattern parse failed: Unicode cathegory expected
        /\p{General_Category=[ HERE ]>>Linear_A}/

/\p{scx}/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\p{scx}/

/\p{gc= Ps}/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /\p{gc=[ HERE ]>> Ps}/

/\P{General_Category=Po }/Ju
    print
    Pattern parse failed: Unknown \p or \P property
        /\P{General_Category=[ HERE ]>>Po }/
